### DISCUSSION TITLE
Typing for Typescript

### DISCUSSION THREAD

**Discussion by DreamTexX** [alpinejs/alpine#1712] [1. Help]
Hi, are there any efforts to make typings for typescript?

Like:
```ts
Alpine.data("foo", () => ({
  foo: "bar",
  init() {
     this.$watch('foo', () => {});
  }
}));
```
Here this.$watch would not be available

**Comment by danddanddand** [comment]
There is a thread about it. I was working on one for my project last week but got distracted by some other stuff. I do plan on making a PR for it eventually tho. https://github.com/alpinejs/alpine/discussions/1609

**Reply by DreamTexX** [reply]
found them thanks, needed to make some changes to make it work in deno but its working without any error now so far.

**Reply by danddanddand** [reply]
Cool. It’s early starting place when we converted to v3. It’s been “good enough” for now but I’m sure developers at my company have added to it and when I get some time would like to make it full definition
